File: 2560_house_robber_iv.java
class Solution {
    public int minCapability(int[] x, int y) {
        int l = 1, h = 1_000_000_000;
        while (l < h) {
            int m = (l + h) / 2, c = 0, i = 0;
            while (i < x.length) {
                if (x[i] <= m) { c++; i++; }
                i++;
            }
            if (c >= y) h = m;
            else l = m + 1;
        }
        return l;
    }
}

File: 2560_house_robber_iv.py
class Solution(object):
    def minCapability(self, nums, k):
        l, r = min(nums), max(nums)
        while l < r:
            m, c, i = (l + r) // 2, 0, 0
            while i < len(nums):
                if nums[i] <= m:
                    c += 1
                    i += 1
                i += 1
            if c >= k:
                r = m
            else:
                l = m + 1
        return l
